teamapps-ux/src/main/java/org/teamapps/ux/resource/Resource.java

```diff
@@ -27,6 +27,9 @@
 import java.util.Date;
 import java.util.function.Supplier;
 
+/**
+ * Interface for handling any type of data source that can be represented as an InputStream.
```

Comment:
This is true. However, the main idea is that Resources are meant to be downloaded by the client using HTTP. Therefore, they provide all the necessary HTTP header information in addition to the data.